### PR TITLE
Partial procedure to deactivate a user

### PR DESCRIPTION
**Pull request by leogrim**
@andrewconner

### CODE REVIEW

**Comment by andrewconner** [comment]
Nice! We've needed this for a long time.

I'm unsure how it'll work if the person tries to rejoin later. Will we find their inactive social user and re-activate it, or create a new one? We definitely want to make sure it doesn't come back... perhaps removing the userId field in their socialuserinfo, removing the network's id, etc.

File: kifi-backend/modules/shoebox/conf/shoebox.routes [hunk 1]
```diff
@@ -354,6 +354,7 @@ POST    /admin/user/:id/update      @com.keepit.controllers.admin.AdminUserContr
 POST    /admin/user/:id/setInvitesCount      @com.keepit.controllers.admin.AdminUserController.setInvitesCount(id: Id[User])
 GET     /admin/user/:id/social_users/refresh @com.keepit.controllers.admin.AdminUserController.refreshAllSocialInfo(id: Id[User])
 POST    /admin/user/:id/impersonate @com.keepit.controllers.admin.AdminAuthController.impersonate(id: Id[User])
+GET     /admin/user/:id/deactivate  @com.keepit.controllers.admin.AdminUserController.deactivate(id: Id[User], readOnly: Boolean = true)
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/conf/shoebox.routes]: Mind changing to a POST? Don't want to accidentally delete users by visiting the page.

Comment (leogrim) [inline kifi-backend/modules/shoebox/conf/shoebox.routes]: readOnly is true by default, not enough?

File: kifi-backend/modules/shoebox/app/com/keepit/admin/AdminUserController.scala [hunk 2]
```diff
@@ -762,4 +762,30 @@ class AdminUserController @Inject() (
...
+      val socialUsers = socialUserInfoRepo.getByUser(userId)
+      val socialConnections = socialConnectionRepo.getSocialConnectionInfosByUser(userId)
+      val userConnections = userConnectionRepo.getConnectedUsers(userId)
+
+      if (!readOnly) {
+        userConnectionRepo.deactivateAllConnections(userId)
+        socialUsers.foreach { sui =>
+          socialConnectionRepo.deactivateAllConnections(sui.id.get)
+          socialUserInfoRepo.save(sui.withState(SocialUserInfoStates.INACTIVE).copy(userId = None, credentials = None, socialId = SocialId.inactive))
+        }
```
Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/admin/AdminUserController.scala:779]: @andrewconner if they can come back in, they deserve it.

File: kifi-backend/modules/shoebox/conf/shoebox.routes (hunk 1 above)
Comment (2is10) [inline kifi-backend/modules/shoebox/conf/shoebox.routes]: Not enough. Let’s abide by HTTP semantics whenever possible.

**Pull request merged**